reactivesocket-cpp/src/tcp/TcpDuplexConnection.h

```diff
@@ -0,0 +1,93 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+#pragma once
+
+#include <folly/ExceptionWrapper.h>
+#include <folly/io/async/AsyncSocket.h>
+#include <folly/io/async/AsyncTransport.h>
+#include <reactivesocket-cpp/src/streams/SmartPointers.h>
+#include "reactivesocket-cpp/src/DuplexConnection.h"
+#include "reactivesocket-cpp/src/Payload.h"
+#include "reactivesocket-cpp/src/ReactiveStreamsCompat.h"
+#include "reactivesocket-cpp/src/mixins/IntrusiveDeleter.h"
+#include <folly/SocketAddress.h>
+#include <folly/io/IOBufQueue.h>
+
+namespace folly {
+    class exception_wrapper;
+}
+
+namespace reactivesocket {
+    class TcpSubscriptionBase: public virtual ::reactivesocket::IntrusiveDeleter,
+                               public ::reactivesocket::Subscription {
+    public:
+        ~TcpSubscriptionBase() = default;
+
+        // Subscription methods
+        void request(size_t n) override {
+            LOG(INFO) << "TcpSubscriptionBase.request " << n;
+        }
+
+        void cancel() override {
+            LOG(INFO) << "TcpSubscriptionBase.cancel";
```

Comment:
I think you should do something here. Just imagine someone calls cancel() and keeps the instance for longer.
In your case you will just ignore the cancel call and will keep sending/receiving data. It feels to me that's not intended.